app/models/registration.rb

```diff
@@ -214,12 +219,17 @@ def record_payment(
     receipt,
     user_id
   )
+    # Don't create multiple registration payments for a manual payment
+    return if receipt.instance_of?(ManualPaymentRecord) &&
+              registration_payments.where(receipt_type: CompetitionPaymentIntegration::INTEGRATION_RECORD_TYPES[:manual]).any?
```

Comment:
Why are you returning even before generating a history entry? This feels like you're short-circuiting too much.

I see that your above comment explains why you're short-circuiting _here_ and not somewhere else in the code, but my question is why you want/need this short-circuiting _at all_.